### PR TITLE
dash/datagrid2-component

### PR DESCRIPTION
**Pull request by ddragula**
Changed DataGrid Component to use the new version of DataGrid.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207629902887005

### CODE REVIEW

File: samples/dashboards/components/climate-all/demo.js [hunk 1]
```diff
@@ -665,44 +665,42 @@ async function setupBoard() {
...
+                        enabled: false
                     },
                     TXC: {
-                        headerFormat: 'Maximal Temperature °C',
+                        headFormat: 'Maximal Temperature °C',
                         cellFormat: '{value:.1f}'
                     },
                     TXF: {
-                        headerFormat: 'Maximal Temperature °F',
+                        headFormat: 'Maximal Temperature °F',
```
Comment (sebastianbochan) [inline samples/dashboards/components/climate-all/demo.js:701]: My first impression is: "headerFormat" sounds more intuitive in comparison with "headFormat".

File: docs/datagrid/general.md [hunk 2]
```diff
@@ -23,71 +23,3 @@ By incorporating this component, users can effectively interact with data in a s
     ```typescript
     import * as DataGrid from '@highcharts/dashboards/datagrid';
     ```
-
```
Comment (sebastianbochan) [inline docs/datagrid/general.md:26]: We should leave the introduction, or add it in the future. What is the reason of removing it?

File: samples/dashboards/demo/accounting/demo.html [hunk 3]
```diff
@@ -11,12 +11,6 @@
     A dashboard presenting sample financial results of a company along with
     predictions, as well as revenue and costs compared to the budget.
 </p>
-<div id="popup">
```
Comment (karolkolodziej) [inline samples/dashboards/demo/accounting/demo.html:14]: Why did you remove it?

File: samples/dashboards/demo/accounting/demo.js [hunk 4]
```diff
@@ -553,103 +542,3 @@ board.then(res => {
...
-    }
-});
-
-document.addEventListener('keydown', e => {
-    if (e.key === 'Escape' && popup.style.display !== 'none') {
-        popup.style.display = 'none';
-    }
-});
-
-async function togglePopup(open) {
```
Comment (sebastianbochan) [inline samples/dashboards/demo/accounting/demo.js:575]: What is the reason of removing popup from demo?

File: test/cypress/data-grid/integration/formatting.cy.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+describe('Remove the dashboard.', () => {
+    before(() => {
+        cy.visit('data-grid/v2/cell-formatting');
```
Comment (sebastianbochan) [inline test/cypress/data-grid/integration/formatting.cy.js:3]: It should not refer to v2 path. All demos should be in the datagrid.

File: test/typescript-karma/Dashboards/Component/datagrid.js [hunk 6]
```diff
@@ -6,9 +6,10 @@ import EditMode from '../../../../code/dashboards/es-modules/masters/modules/lay
 Dashboards.DataGridPlugin.custom.connectDataGrid(DataGrid);
 Dashboards.PluginHandler.addPlugin(Dashboards.DataGridPlugin);
 
-const { test } = QUnit;
+const { skip } = QUnit;
 
-test('DataGrid component with dataTable', async function (assert) {
+// DataGrid1 Tests to update
+skip('DataGrid component with dataTable', async function (assert) {
```
Comment (sebastianbochan) [inline test/typescript-karma/Dashboards/Component/datagrid.js:12]: What is the reason of skipping test?

File: test/typescript-karma/Dashboards/Synchronization/sync.js [hunk 7]
```diff
@@ -10,10 +10,11 @@ Dashboards.DataGridPlugin.custom.connectDataGrid(DataGrid);
 Dashboards.PluginHandler.addPlugin(Dashboards.HighchartsPlugin);
 Dashboards.PluginHandler.addPlugin(Dashboards.DataGridPlugin);
 
-const { test } = QUnit;
+const { test, skip } = QUnit;
 
 
-test('Sync events leak in updated components', async function (assert) {
+// DataGrid1 Tests to update
+skip('Sync events leak in updated components', async function (assert) {
```
Comment (sebastianbochan) [inline test/typescript-karma/Dashboards/Synchronization/sync.js:17]: The same as above.

File: test/typescript-karma/Dashboards/Component/highcharts.js [hunk 8]
```diff
@@ -760,7 +760,8 @@ test('JSON data with columnNames and columnAssignment.', async function (assert)
 
 });
 
-test('Crossfilter with string values', async function (assert) {
+// DataGrid1 Tests to update
+skip('Crossfilter with string values', async function (assert) {
```
Comment (sebastianbochan) [inline test/typescript-karma/Dashboards/Component/highcharts.js:764]: The same as above

File: samples/data-grid/basic/cell-events/demo.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            price: [1.5, 2.53, 5, 4.5],
+            metaData: ['a', 'b', 'c', 'd'],
+            icon: ['Apples URL', 'Pears URL', 'Plums URL', 'Bananas URL']
+        }
+    },
+    caption: {
+        text: 'Open the console to see the events being logged.'
+    },
+    events: {
+        cell: {
```
Comment (karolkolodziej) [inline samples/data-grid/basic/cell-events/demo.js:16]: Shouldn't we have a similar for the `head`?

File: ts/Dashboards/Components/DataGridComponent/DataGridSyncs/DataGridHighlightSync.ts [hunk 10]
```diff
@@ -58,21 +58,21 @@ const syncPair: Sync.SyncPair = {
...
             if (table) {
-                const row = e.row;
+                const cell = e.target;
 
                 cursor.emitCursor(table, {
                     type: 'position',
-                    row: parseInt(row.dataset.rowIndex, 10),
-                    column: e.columnName,
+                    row: parseInt(cell?.row?.index, 10),
+                    column: cell?.column?.id,
```
Comment (sebastianbochan) [inline ts/Dashboards/Components/DataGridComponent/DataGridSyncs/DataGridHighlightSync.ts:69]: What in the case when row/column will be undefined? IMO we should not call the emitCursor.

File: ts/DataGrid/DataGridCell.ts [hunk 11]
```diff
@@ -119,40 +117,48 @@ class DataGridCell {
...
+     */
+    public setValue(value: DataTable.CellType, updateTable: boolean): void {
+        const element = this.htmlElement;
 
-        this.value = this.column.data[this.row.index];
+        this.value = value;
 
-        if (this.value) {
-            cellContent = this.formatCell(this.value, this);
+        if (value === null || value === void 0) {
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGridCell.ts:142]: You can use `defined` method from utils.

File: samples/data-grid/basic/column-options/demo.css [hunk 12]
```diff
@@ -0,0 +1,15 @@
+@import url("https://code.highcharts.com/datagrid/css/datagrid.css");
```
Comment (karolkolodziej) [inline samples/data-grid/basic/column-options/demo.css:1]: This might be my mistake but we should use `dashbaords/css/datagrid.css` in the public demo. The only exceptions are the cypress demos which now also should work with `dasboards...` I presume 🤔

File: samples/data-grid/basic/column-resizing-disabled/demo.css [hunk 13]
```diff
@@ -0,0 +1,14 @@
+@import url("https://code.highcharts.com/datagrid/css/datagrid.css");
```
Comment (karolkolodziej) [inline samples/data-grid/basic/column-resizing-disabled/demo.css:1]: Check globally.

File: samples/dashboards/components/climate-all/demo.js (hunk 1 above)
Comment (ddragula) [reply samples/dashboards/components/climate-all/demo.js:701]: But it's a `head`, not a `header`.

File: docs/datagrid/general.md (hunk 2 above)
Comment (ddragula) [reply docs/datagrid/general.md:26]: I guess it's a part of the documentation task.

File: samples/dashboards/demo/accounting/demo.js (hunk 4 above)
Comment (ddragula) [reply samples/dashboards/demo/accounting/demo.js:575]: It didn't work for some time and the data presented there wasn't important/easily readable.

File: ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts [hunk 14]
```diff
@@ -129,338 +137,88 @@ class DataGridComponent extends Component {
...
-            ...base,
-            options: {
-                ...base.options,
-                dataGridOptions
-            }
-        };
-
-        this.emit({ type: 'toJSON', json });
-        return json;
+    public onTableChanged(): void {
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts:209]: Add docklet

File: test/cypress/data-grid/integration/formatting.cy.js (hunk 5 above)
Comment (ddragula) [reply test/cypress/data-grid/integration/formatting.cy.js:3]: Yes, but the test updating is done in a different task.

File: test/typescript-karma/Dashboards/Component/datagrid.js (hunk 6 above)
Comment (ddragula) [reply test/typescript-karma/Dashboards/Component/datagrid.js:12]: Tests for the DataGrid Component are also part of this task: https://app.asana.com/0/1203234853779882/1207620146790059/f

File: test/typescript-karma/Dashboards/Synchronization/sync.js (hunk 7 above)
Comment (ddragula) [reply test/typescript-karma/Dashboards/Synchronization/sync.js:17]: Tests for the DataGrid Component are also part of this task: https://app.asana.com/0/1203234853779882/1207620146790059/f

File: ts/DataGrid/.eslintrc [hunk 15]
```diff
@@ -1,10 +1,13 @@
 {
     "extends": "../.eslintrc",
     "rules": {
-      "jsdoc/require-param": 0,
-      "jsdoc/require-param-description": 0,
+      "jsdoc/require-jsdoc": 2,
```
Comment (karolkolodziej) [inline ts/DataGrid/.eslintrc:4]: Do we need it? If so, let's keep it inline with what we have for Dashboards.

File: ts/DataGrid/Actions/CellEditing.ts [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+        this.renderInput();
+    }
+
+    /**
+     * Stops the editing of the cell.
+     *
+     * @param submit
+     * Whether to save the value of the input to the cell. Defaults to true.
+     */
+    public stopEditing(submit = true): void {
```
Comment (karolkolodziej) [inline ts/DataGrid/Actions/CellEditing.ts:97]: It is worth mentioning that in the docs article which key does what

File: samples/data-grid/basic/cell-events/demo.js (hunk 9 above)
Comment (ddragula) [reply samples/data-grid/basic/cell-events/demo.js:16]: Maybe, but I would treat it as an idea for the future.

File: ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts (hunk 14 above)
Comment (ddragula) [reply ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts:209]: It's an overrided method, so it's not needed here. I'll add the `override` keyword to make it more clear.

File: ts/DataGrid/.eslintrc (hunk 15 above)
Comment (ddragula) [reply ts/DataGrid/.eslintrc:4]: It just helps with catching mistakes in doclets. Can be changed also in Dashboards, but I would do it in a different task when we'll be working on the new API docs.

File: samples/dashboards/components/climate-all/demo.js (hunk 1 above)
Comment (sebastianbochan) [reply samples/dashboards/components/climate-all/demo.js:701]: @karolkolodziej what do you think?

Comment (ddragula) [reply samples/dashboards/components/climate-all/demo.js:701]: We need to decide on a uniform nomenclature. If we decide on `header` instead of `head`, we need to change it everywhere.

Comment (karolkolodziej) [reply samples/dashboards/components/climate-all/demo.js:701]: Technically we are not styling `<thead>` but its child `<th>` which stands for the header. So in this case maybe it makes more sense for `headerFormat`? But the whole structure should still be called `head`

**Review by karolkolodziej — APPROVED** [review]
